portfolio/src/main/java/com/google/sps/servlets/DeleteCommentServlet.java

```diff
@@ -0,0 +1,32 @@
+package com.google.sps.servlets;
+
+import com.google.gson.Gson;
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.Key;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
+import java.io.IOException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.util.ArrayList;
+import java.lang.Integer;
+
+@WebServlet("/delete-comment")
+public class DeleteCommentServlet extends HttpServlet {
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+      Query query = new Query("Comment");
+      
+      DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+      PreparedQuery results = datastore.prepare(query);
+      for (Entity entity : results.asIterable()) {
+        datastore.delete(entity.getKey());    
```

Comment:
A good practice is for each RPC that does a mutation, that it should be transactional Consider the case where only some of the deletes succeed.  The behavior would be an error is reported and then when you refresh the page only some of them will have been deleted.   Your options are:

1) Change your backend (servlet) logic that only allows 1 delete per HTTP request.  (The UI would need to change to accommodate this.)
2) Use DB transactions to make the transaction atomic
3) Make it so if some succeed or some fail, that you report back to the client which failed and which succeed.  (This is kind of ugly, and I dont' recommend that.)

Since this is a starter project, I wouldn't worry about fixing it, but in your capstone project you'd want to design things that way.